File: December 2022/Day 17/C++ Code/Break a number.cpp
// Problem Link :- https://practice.geeksforgeeks.org/problems/break-a-number5913/1
// Video Solution Link :- https://youtu.be/gOvj1bBg-VI

//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++
class Solution{   
public:
    // Time Complexity :- O(1);
    // Space Complexity :- O(1);
    int waysToBreakNumber(int N){
        // code here
        long long int mod = 1000000007;
        long long int n =N;
        return (3*n+((n-1)*(n-2))/2)%mod;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N;
        cin >> N;
        Solution ob;
        cout << ob.waysToBreakNumber(N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: December 2022/Day 17/JAVA Code/Break a number.java
// Problem Link :- https://practice.geeksforgeeks.org/problems/break-a-number5913/1
// Video Solution Link :- https://youtu.be/gOvj1bBg-VI

//{ Driver Code Starts

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            int N = Integer.parseInt(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.waysToBreakNumber(N));
        }
    }
}
// } Driver Code Ends



class Solution{
    // Time Complexity :- O(1);
    // Space Complexity :- O(1);
    private static final long mod = 1000000007;
    int waysToBreakNumber(int N){
        // code here
        long n = N;
        return (int)((3*n+((n-1)*(n-2))/2)%mod);
    }
}
